### ISSUE TITLE
Unable to rename security group (resource has a dependent object)

### ISSUE THREAD

**Issue by ndench** [terraform-aws-modules/terraform-aws-security-group#39]
I'm trying to rename my security group so that I know what environment it's for:
```hcl
module "security_group_web" {
  source  = "terraform-aws-modules/security-group/aws"
  version = "1.15.0"

  name        = "web"
...
}
```

```hcl
module "security_group_web" {
  source  = "terraform-aws-modules/security-group/aws"
  version = "1.15.0"

  name        = "staging-web"
...
}
```

However I get this error:
```
Error: Error applying plan:

1 error(s) occurred:

* module.staging.module.security_group_web.aws_security_group.this (destroy): 1 error(s) occurred:

* aws_security_group.this: DependencyViolation: resource sg-c5a360bc has a dependent object
        status code: 400, request id: 5f113cf5-8d37-4587-9797-c0b24e349936
```

I assume this is because the all the `security_group_rule` resources that depend on the `security_group`? Is there a way around this?

**Comment by tchia04** [comment]
I believe there are probably some EC2 instances that is using the web security group.  If you tries to delete the "web" security group, you will probably get the same/similar error message about dependent objects.

**Comment by ndench** [comment]
@tchia04 you are correct. I thought I checked everything but I had a random snowflake instance lying around that was using the security group. Once I killed that then everything seems to work.

The only problem is that I have a load balancer defined in my terraform config that uses the security group, and terraform still blocks me from renaming it. So I first have to remove the security group from the load balancer, run `terraform apply` then rename the security group and add it back to the load balancer and run `terraform apply` again.

But that's fine, it's kind of expected anyway. Thanks for your help :smile:
